src/device/index.js

```diff
@@ -392,6 +406,74 @@ realityEditor.device.postEventIntoIframe = function(event, frameKey, nodeKey) {
             }
         }
     }
+    
+    const entries = [];
+    realityEditor.forEachFrameInAllObjects((object, frame) => {
+        entries.push({
+            object,
+            frame
+        });
+    });
+    let currentIndex = realityEditor.device.raycastIndex;
+    realityEditor.device.raycastIndex += 1;
+    const frameRaycastPromises = entries.map(entry => {
+        return new Promise((resolve, reject) => {
+            const {object, frame} = entry;
+            const targetIframe = document.getElementById('iframe' + frame);
+            if (!targetIframe || !targetIframe.contentWindow) { // Avatars don't have iframes
+                resolve(null);
+                return;
+            }
+            targetIframe.contentWindow.postMessage(JSON.stringify({
+                raycastRequest: {
+                    coords: {
+                        x: newCoords.x,
+                        y: newCoords.y
+                    },
+                    index: currentIndex
+                }
+            }), '*');
+            let resolved = false;
+
+            const key = object + frame + currentIndex;
+            setTimeout(() => {
+                realityEditor.device.raycastCallbacks.delete(key);
+                if (!resolved) {
+                    resolved = true;
+                    reject();
+                }
+            }, 200); // Reject automatically within 200 ms
+
+            realityEditor.device.raycastCallbacks.set(key, (result) => {
+                if (!resolved) {
+                    resolved = true;
+                    resolve(result);
+                }
+            });
+        })
+    });
+    const raycastResults = await Promise.all(frameRaycastPromises);
```

Comment:
Just leave it here for potential consideration: this line alone spends ~2-4 ms of our performance budget (~16.7 ms in total if we're to maintain 60 fps). So we could potentially have a flag in device/index.js that gets set when one tool has enabled drawing / raycasting inside of it, and if no tool is enabled, we can skip this entire section.

Reply:
It's async, so unless I'm mistaken, this is just a 2-4ms latency between a mouse click and the tool receiving the event info. It should not be blocking the event loop/taking up our performance budget.

Reply:
Oh make sense